system/src/system_control_internal.cpp

```diff
@@ -389,6 +415,44 @@ void system_ctrl_set_result(ctrl_request* req, int result, ctrl_completion_handl
     particle::system::SystemControl::instance()->setResult(req, result, handler, data);
 }
 
+int system_ctrl_add_request_filter(system_control_acl default_action, system_control_filter* filters, void* reserved) {
+    if (default_action == SYSTEM_CONTROL_ACL_NONE) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+    particle::system::SystemControl::instance()->filterCtrlReq_.vecCtrlReq_.clear();
```

Comment:
I think we've discussed the same thing in a different context: functions should not access internal members of classes. Add an accessor method to the class instead.